docker/ubuntu/baseimage-clang/Dockerfile

```diff
@@ -1,17 +1,17 @@
-FROM koreader/kobase:0.2.2
+FROM koreader/kobase:0.3.0
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 USER root
-COPY clang12.list /etc/apt/sources.list.d/clang12.list
-RUN apt-get remove gcc gcc-8 -y \
+COPY clang16.list /etc/apt/sources.list.d/clang16.list
+RUN apt-get remove gcc gcc-9 gcc-10 -y \
     && wget --progress=dot:giga -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - \
     && apt-get update \
-    && apt-get install -y --no-install-recommends clang-12 \
-    && update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 380 \
-    && update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-12 380 \
-    && update-alternatives --install /usr/bin/clang-cpp clang-cpp /usr/bin/clang-cpp-12 380 \
-    && update-alternatives --install /usr/bin/gcc gcc /usr/bin/clang-12 380 \
+    && apt-get install -y --no-install-recommends libtool-bin clang-16 \
```

Comment:
Not really sure why only clang but either way, also see `https://github.com/zeromq/libzmq/issues/1385`